File: packages/models/lib/src/widgets/features/text_direction.dart
// Package imports:
// ignore_for_file: public_member_api_docs

// Package imports:

import 'package:enum_to_string/enum_to_string.dart';
import 'package:flutter/material.dart';
import 'package:theta_models/src/models/feature.dart';

@immutable
class FTextDirection extends FeatureWithEnum<TextDirection> {
  const FTextDirection({
    super.value = TextDirection.ltr,
  });

  static FTextDirection fromJson(final String json) => FTextDirection(
      value: EnumToString.fromString(TextDirection.values, json)!);
}

File: figma_plugin/src/features/text_direction.ts
export class FTextDirection {
  value: TextDirection;

  constructor({ value = TextDirection.ltr }: { value?: TextDirection }) {
    this.value = value;
  }

  toJson(): string {
    return this.value;
  }
}

export enum TextDirection {
  ltr = "ltr",
  rtl = "rtl"
}
